apps/core/scenes/search/Search.js

```diff
@@ -58,39 +70,54 @@ class Search extends React.Component<Props> {
     if (Array.isArray(locations)) {
       return locations.reduce((acc, location, index) => {
         const prefix = index > 0 ? ',' : '';
-        return `${acc}${prefix}${location[field]}`;
+        const locationField = location[field] ?? '';
+        return `${acc}${prefix}${locationField}`;
       }, '');
     }
     return '';
   };
 
   handleSubmitPress = () => {
-    const {
-      travelFrom,
-      travelTo,
-      dateFrom,
-      dateTo,
-      returnDateFrom,
-      returnDateTo,
-      tripType,
-    } = this.props;
+    const { travelFrom, travelTo, tripType, onSubmit } = this.props;
     if (travelFrom == null) {
       this.props.setAlertContent({
         message: 'Please fill this form completely before you proceed',
       });
     } else {
+      const dateFrom = format(this.props.dateFrom, BASIC_ISO_DATE_FORMAT);
+      const dateTo = format(this.props.dateTo, BASIC_ISO_DATE_FORMAT);
+      const returnDateFrom = format(
+        this.props.returnDateFrom,
+        BASIC_ISO_DATE_FORMAT,
+      );
+      const returnDateTo = format(
+        this.props.returnDateTo,
+        BASIC_ISO_DATE_FORMAT,
+      );
+      if (onSubmit != null) {
+        onSubmit({
```

Comment:
So this will only be executed on web for now, since there is no onSubmit on mobile, right?

Reply:
Yes, and the context is preserved on mobile, so no need to do it any other way there.